### ISSUE TITLE
Syntaxe Issue when a value begins with :

### ISSUE THREAD

**Issue by neooblaster** [se38/zJSON#31]
Hello Uwe Fetzer,

I encounter an issue with the library ZCL_JSON_DOCUMENT.
I'm not able to said which version / how old the installation is (I'm working in AMS Team) but I checked in your `src` folder for method `get_json_large` and I found the REGEXP which break the JSON syntaxe.

In my installation, I have only `get_json` but in master version, the `get_json` call `get_json_large`.
So I warn you, maybe your library contain the following error : 

- When my field value begins with `:`, the JSON syntaxe is broken : 

![image](https://user-images.githubusercontent.com/22078558/115428809-13757b00-a203-11eb-9471-d0b3065fc6ac.png)

You can easily reproduce the error and fix the issue (if it still exists in the lastest version).

For the story : 

In our usage case, we use the library to generates a JSON answer to a Fiori application (throught a Web Service).
JavaScript will no be able to parse the answer and stop working. 
It's blocking for the user.

Action : 

I can make the correction on our installation, but I have no idea what is the purpose of the fix `sapcodexch issue #3`.
Can I removed the REGEXP check ?

Last question, is `*` in REGEXP stands for "all" ? Maybe I can replace it by `"(.*)":` ?

Our version :
![image](https://user-images.githubusercontent.com/22078558/115431117-3d2fa180-a205-11eb-94fb-2e79c72839fa.png)

Your version :
![image](https://user-images.githubusercontent.com/22078558/115431051-2db05880-a205-11eb-83d4-a4e32ce69ec8.png)

Thanks a lot for your help.
Kindly regards,

Nicolas DUPRE.

````abap
TYPES: BEGIN OF ty_json,
         matnr TYPE matnr,
         maktx TYPE maktx,
       END   OF ty_json.

DATA : lo_json TYPE REF TO zcl_json_document.
DATA : lo_json2 TYPE REF TO zcl_json_document.
DATA : lt_data TYPE TABLE OF ty_json.
DATA : lt_data2 TYPE TABLE OF ty_json.
DATA : lv_json TYPE string.
DATA : lv_json2 TYPE string.

APPEND VALUE #(
  matnr = 'A12345'
  maktx = 'Matnr desc'
) TO lt_data.
APPEND VALUE #(
  matnr = 'A12345'
  maktx = ':Matnr desc'
) TO lt_data2.

CREATE OBJECT lo_json.
CREATE OBJECT lo_json2.

lo_json->set_data(
  suppress_itab = 'X'
  data          = lt_data
).
lo_json2->set_data(
  suppress_itab = 'X'
  data          = lt_data2
).

lv_json = lo_json->get_json( ).
lv_json2 = lo_json2->get_json( ).

WRITE : lv_json. WRITE : / . WRITE : lv_json2.
````

**Comment by se38** [comment]
Hi Nicolas,
thank you for your report. Will have a look later today.
Cheers, Uwe

**Comment by se38** [comment]
Hi Nicolas,

you are definately on an older version, but I could reproduce the error.
I've fixed the error in my system (and will upload a new versioner later).

In you version try the following:
After 
```
IF me->json IS INITIAL.
ELSE.
```
add the following line:
`SHIFT me->json LEFT DELETING LEADING space.`

Please report back if this works for you.
Uwe

**Comment by neooblaster** [comment]
Hello Uwe,

I tried the suggested correction and I'm happy to announce that works perfectly : 

![image](https://user-images.githubusercontent.com/22078558/115563955-547b9700-a2b8-11eb-9de0-16e410579351.png)

The execution of the same test program next to the correction  : 

![image](https://user-images.githubusercontent.com/22078558/115564077-6c531b00-a2b8-11eb-9441-288b6727aae4.png)


Thanks a lot for this very quick answer and fix proposal.

I close the issue or you prefer to close it to a final commit ?

Kindly regads,
Nicolas D.

**Comment by se38** [comment]
Glad it worked :-)
Closed with version 2.35
